modules/ROOT/pages/_partials/commons/common-new-logging-api.adoc

```diff
@@ -0,0 +1,634 @@
+
+----
+
+
+.New API
+:!example-caption:
+[source, c, indent=0]
+----
+
+include::c:example$code_snippets/main.cpp[tags=new-console-logging]
+
+----
+endif::is-c[]
+
+[#lbl-file-logsink]
+=== Logging to the Couchbase File Log
+
+The changes necessary to convert the installation of a file logger are also similar. 
+Instead of configuring a `FileLogger` using a`LogFileConfiguration`, create a new `FileLogSink` with the desired properties and install it.
+
+NOTE: `setRotateCount` from the old API is slightly different from `setMaxKeptFiles`.
+`setMaxKeptFiles` is the maximum number of log files that will exist at any time and is the count of rotated files (setRotateCount) plus one.
+
+
+
+ifdef::is-android[]
+
+[tabs]
+=====
+Java::
++
+--
+
+.Old API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+--
+
+Kotlin::
++
+--
+
+.Old API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, kotlin, indent=0]
+----
+
+include::android:example$kotlin_snippets/app/src/main/kotlin/com/couchbase/codesnippets/BasicExamples.kt[tags=new-file-logging]
+
+----
+--
+=====
+
+endif::is-android[]
+ifdef::is-java[]
+
+.Old API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, java, indent=0]
+----
+
+include::java:example$snippets/common/main/java/com/couchbase/codesnippets/Examples.java[tags=new-file-logging]
+
+----
+
+endif::is-java[]
+ifdef::is-csharp[]
+
+.Old API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=file-logging]
+----
+
+.New API
+:!example-caption:
+[source, csharp, indent=0]
+----
+include::csharp:example$code_snippets/Program.cs[tags=new-file-logging]
+----
+endif::is-csharp[]
+ifdef::is-objc[]
+
+.Old API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, objc, indent=0]
+----
+
+include::objc:example$code_snippets/SampleCodeTest.m[tags=new-file-logging]
+
+----
+endif::is-objc[]
+ifdef::is-swift[]
+
+.Old API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=file-logging]
+
+----
+
+.New API
+:!example-caption:
+[source, swift]
+----
+
+include::swift:example$code_snippets/SampleCodeTest.swift[tags=new-file-logging]
```

Comment:
I need to make an update to this - to add `maxSize`, such that is matching 1to1 with the old snippet.

Reply:
Okay, waiting for this update